File: 백준/Gold/2170. 선 긋기/선 긋기.py
import sys, collections, heapq, math, re, random
input=sys.stdin.readline
N = int(input())
arr = []
for _ in range(N):
    x,y = map(int,input().split())
    x,y = min(x,y),max(x,y)
    arr.append((x,y))
arr.sort()
ans=0
l,r=arr[0]
for x,y in arr:
    if r<x:
        ans+=(r-l)
        l,r = x,y
    else:
        r=max(r,y)
ans+=(r-l)
print(ans)

File: 백준/Gold/2170. 선 긋기/선 긋기.java
import java.util.*;
import java.io.*;

class Point {
    long x, y;

    Point(long x, long y) {
        this.x = x;
        this.y = y;
    }

}

public class Main {
    public static void main(String args[]) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        ArrayList<Point> arr = new ArrayList<Point>();
        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            long a = Long.parseLong(st.nextToken());
            long b = Long.parseLong(st.nextToken());
            arr.add(new Point(a, b));
        }
        arr.sort(new Comparator<Point>() {
            @Override
            public int compare(Point p1, Point p2) {
                if (p1.x > p2.x)
                    return 1;
                else if (p1.x == p2.x && p1.y > p2.y)
                    return 1;
                else
                    return -1;
            }
        });
        int dist = 0;
        Point ans = arr.get(0);
        for (Point p : arr) {
            if (ans.y < p.x) {
                dist += ans.y - ans.x;
                ans = p;
            } else {
                ans.y = Math.max(ans.y, p.y);
            }
        }
        dist += ans.y - ans.x;
        System.out.println(dist);
    }
}